core/ecschema-editing/src/Differencing/SchemaDifferenceValidator.ts

```diff
@@ -231,25 +212,20 @@ class SchemaDifferenceValidationVisitor implements ISchemaDifferenceVisitor {
    * @internal
    */
   public async visitEntityClassMixinDifference(entry: EntityClassMixinDifference) {
-    const targetSchemaClass = await this.getSchemaItem<EntityClass>(entry.itemName);
+    const targetEntityClass = await this._targetSchema.getItem(entry.itemName) as EntityClass;
     for (const addedMixin of entry.difference) {
       // To validate the added mixins, the instance from the source schema it fetched,
       // otherwise validation gets too complicated as the mixin must not be existing in
       // the current target schema, it could also be added to the schema.
-      const sourceMixin = await this._sourceSchema.lookupItem<Mixin>(addedMixin);
-      const sourceSchemaItem = await this._sourceSchema.getItem<EntityClass>(entry.itemName);
-      if (sourceMixin === undefined) {
-        return;
-      }
-
-      const appliesTo = sourceMixin.appliesTo as Readonly<SchemaItemKey>;
-      if (appliesTo && !await derivedFrom(sourceSchemaItem, appliesTo.name)) {
+      const sourceMixin = await this._sourceSchema.lookupItem(addedMixin) as Mixin;
+      const sourceSchemaItem = await this._sourceSchema.getItem(entry.itemName) as EntityClass;
+      if (sourceMixin.appliesTo && !await this.derivedFrom(sourceSchemaItem, sourceMixin.appliesTo)) {
```

Comment:
I don't know why this needs any special code.  Both classes are from the source schema so you should be able to use ECClass.is